### ISSUE TITLE
Merge regulation.bin without CSV (revisited)

### ISSUE THREAD

**Issue by pokkla** [vawser/Smithbox#156]
In the issue "Merge regulation.bin without CSV file #107" it was mentioned that a regulation.bin could be merged into the base regulation.bin file.  The reply included "You can then repeat this process for each mod you want to merge in."  This is my preferred method to merge or re-merge many regulation.bin files as I add/remove mods.  It saves having to merge lots of CSV files.  [BTW, I use MO2 from https://github.com/ModOrganizer2/modorganizer.]  However, I had a potentially awful thought with the merging.  That is, writing params back to original values without knowing it.

I understand the target regulation.bin merges with the current project changing "only unique params".  This is fine for the first target regbin merged in.  But, what happens with the second, third, etc regbin merged in?  Does "unique param" compare to the base game regbin, or to the regbin in the project?  If base game, great and edits will be additive over time.  However, if unique to the project regbin, then I could be inadvertently overwriting previously changed params back to base game params, if they were not changed for the second, third, regbin mod.

Does that make sense?

**Comment by pokkla** [comment]
After additional testing, here are my conclusions.
1) Smithbox only merges in values that differ from the "vanilla bank" of values.  The comparison is NOT with the current value in the current regbin.  At this time, the vanilla values are from v1.16.0.1000 of the regulation.bin (regbin).
2) You can merge in 2nd, 3rd, 4th, etc regbin files.  But, you must save (modified params) and exit smithbox, then reload the new regbin between merges.  If you try to merge several regbins in a row, without saving/exiting between, only the first merged values will be retained.  [Note: There may also be a menu or hotkey I'm missing to avoid save/exit smithbox.]
3) If a subsequent regbin replaces the exact param, line AND field, the latter regbin will overwrite the previously changed value.  This, most of us, would expect to happen.  But, if any ONE of those (param, line, field) are different, then smithbox considers it a whole different parameter.
4) There appears to be no mechanism to remove just the changed values.  So, if the base game (vanilla) regbin has 0 for a value on something, and you merge in, say, a value of 10, there is no automated way, to change just that one back to default of 0.  Of course, you can change any value manually.  But I found it easier to re-merge regbins without including the mod I'm removing.

In summary, I think the quickest method:
Merge1 > Save > Exit > Load new > Merge2 > Save > Exit > Load new-new > Merge3 > Save (etc.)

If you do it this way, all the changed values in all the merged regbins should be in the final regulation.bin file.  Plus, there is no need to use any CSV files for any of the parameters.

Let me know if you see any errors in my analysis.  Peace, out.
(I'll hold off on closing this issue for a couple weeks to see if anyone else comments)
